web/src/scheduled_messages.js

```diff
@@ -315,3 +317,29 @@ export function initialize(scheduled_messages_params) {
         e.stopPropagation();
     });
 }
+
+// exporting for testability
+export const _test_should_update_send_later_options = should_update_send_later_options;
+function should_update_send_later_options(date) {
+    const current_minute = date.getMinutes();
+    const current_hour = date.getHours();
+    // We need to rerender if it is either 5 minutes before the hour or at midnight.
+    return current_minute === 55 || (current_hour === 0 && current_minute === 0);
+}
+
+function is_send_later_modal_open() {
+    return overlays.is_modal_open() && overlays.active_modal() === "#send_later_modal";
+}
+
+function update_send_later_options() {
+    const now = new Date();
+    const is_need_to_rerender = is_send_later_modal_open() && should_update_send_later_options(now);
+    if (is_need_to_rerender) {
+        const filtered_send_opts = get_filtered_send_opts(now);
+        $("#send_later_modal .send_later_options").replaceWith(
+            render_send_later_modal_options(filtered_send_opts),
+        );
+    }
+}
+
+setInterval(update_send_later_options, 1000 * 60);
```

Comment:
> I wonder if it's worth setting the interval only when the modal opens

That will definitely be worth it! It will help eliminate the need for `is_send_later_modal_open`, at the very least. Thanks!

> you could have a const like SCHEDULING_MODAL_UPDATE_INTERVAL_IN_MILLISECONDS

Great, thanks!